components/board/pinwrappers/analogs.go

```diff
@@ -104,6 +110,14 @@ func (as *AnalogSmoother) Start() {
 	as.workers = utils.NewStoppableWorkers(func(ctx context.Context) {
 		consecutiveErrors := 0
 		var lastError error
+
+		// Store the full analog reader value
+		analogVal, err := as.Raw.Read(ctx, nil)
```

Comment:
I think there's a race condition here: if someone calls `Read()` before the background goroutine has gotten to here, the min/max/stepsize will all be 0. I suspect this paragraph should happen in `SmoothAnalogReader()` instead of here, so the entire struct is initialized before it can be used.